js/interactive-guides/utils.js

```diff
@@ -40,69 +40,94 @@ var utils = (function() {
+            }
+            // check if contain quote
+            if (title.indexOf("\"") !== -1 ||
+                title.indexOf("'") !== -1) {
+                //var quote = title.substring(0, 1);
+                //var tmpString = title.substring(1);
+                //console.log("tmpString ", tmpString);
+                //var lastIndex = tmpString.indexOf(quote);
+                //title = title.substring(0, lastIndex + 2);
+            } else { 
+                title = "'" + title + "'";
+            }
         }
-        return name;
+        //console.log("title=", title);
+        return title;
     };
     
     var __getCallbackAction = function(strAction) {
-        //console.log("BBB string ", strAction);
-        var callbackStr;
-        if (strAction.indexOf("callback=") !== -1) {
-            var index = strAction.indexOf("callback=") + 9;
-            //console.log("index ", index);
+        var callbackStr, index;
+           
+        if (strAction.indexOf("onclick=") !== -1) {
+            index = strAction.indexOf("onclick=") + 8;
+        } else if (strAction.indexOf("onkeypress=") !== -1) {
+            index = strAction.indexOf("onkeypress=") + 11;
+        }
+        //console.log("index ", index);
+        if (index) {
             callbackStr = strAction.substring(index);
             //console.log("callback ", callbackStr);
             var lastIndex = callbackStr.indexOf(")");
-            callbackStr = callbackStr.substring(0, lastIndex + 2);
-            //console.log("callback=", callbackStr);
+            var hasQuotes = callbackStr.substring(0, 1);
+            if (hasQuotes.indexOf("\"") !== -1 ||
+                hasQuotes.indexOf("'") !== -1) {
+                callbackStr = callbackStr.substring(0, lastIndex + 2);
+            } else {
+                callbackStr = "\"" + callbackStr.substring(0, lastIndex + 1) + "\"";
+            }
         }
+        //console.log("callback=", callbackStr);
         return callbackStr;
     };
     
     var __getButtonName = function(strName) {
-        var buttonName = strName.substring(1, strName.length - 1);
-        var firstIndex = strName.indexOf("<b>") + 3;
-        var lastIndex = strName.indexOf("</b>");
-        var buttonName = strName.substring(firstIndex, lastIndex);
-        //console.log("buttonName ", buttonName);
+        var buttonName;
+        if (strName.indexOf("<b>") !== -1) {
+            var firstIndex = strName.indexOf("<b>") + 3;
+            var lastIndex = strName.indexOf("</b>");
+            buttonName = strName.substring(firstIndex, lastIndex);
+        } else {
+            //get string from last index of <action forward
+            var tmpStr = strName.substring(7);
+            if (tmpStr.indexOf(">") !== -1) {
+                var firstIndex = tmpStr.indexOf(">") + 1;
+                var lastIndex = tmpStr.indexOf("</action>");
+                buttonName = tmpStr.substring(firstIndex, lastIndex);
+            }
```

Comment:
Don't need for the if part logic. The else part logic should take care of getting the button string.